enterprise/app/controllers/enterprise/webhooks/stripe_controller.rb

```diff
@@ -18,4 +27,23 @@ def process_payload
     # We've successfully processed the event without blowing up
     head :ok
   end
+
+  private
+
+  def determine_webhook_secret(payload)
+    # Parse the payload to check event type without full verification
+    parsed_payload = JSON.parse(payload)
+    event_type = parsed_payload['type']
+
+    if v2_billing_event?(event_type)
+      ENV.fetch('STRIPE_WEBHOOK_SECRET_V2', nil)
```

Comment:
@sojan-official should we move this to a installation config? and eventually remove `STRIPE_WEBHOOK_SECRET` from the ENV

@tds-1 let Sojan confirm before making this change